frontend/packages/ux-editor/src/components/config/editModal/EditDataModelBindings.test.tsx

```diff
@@ -75,28 +85,195 @@ const render = async ({ dataModelBindings = {}, handleComponentChange = jest.fn(
     ).toBeInTheDocument();
     expect(await screen.findByText('testModel.field1')).toBeInTheDocument();
   });
+
+  it('should render link icon', async () => {
+    await render();
+    const linkIcon = screen.getByText(textMock('ux_editor.modal_properties_data_model_link'));
+    expect(linkIcon).toBeInTheDocument();
+  });
+
+  it('should show select when link icon is clicked', async () => {
+    await render();
+    const linkIcon = screen.getByText(textMock('ux_editor.modal_properties_data_model_link'));
+    act(() => {
+      linkIcon.click();
+    });
+    const select = screen.getByRole('combobox');
+    expect(select).toBeInTheDocument();
+  });
+
+  it('should toggle select on link icon click', async () => {
+    await render();
+    expect(screen.queryByRole('combobox')).not.toBeInTheDocument();
+    const linkIcon = screen.getByText(/ux_editor.modal_properties_data_model_link/i);
+    act(() => {
+      linkIcon.click();
+    });
+    expect(screen.getByRole('combobox')).toBeInTheDocument();
+  });
+
+  it('check that handleDataModelChange is called', async () => {
+    const handleDataModelChange = jest.fn();
+    const dataModelSelectVisible = jest.fn();
+    await render({ handleDataModelChange });
+    const linkIcon = screen.getByText(/ux_editor.modal_properties_data_model_link/i);
+    act(() => {
+      linkIcon.click();
+    });
+    dataModelSelectVisible(true);
+    const select = screen.getByRole('combobox');
+    const option = within(select).getByText('');
+    act(() => {
+      option.click();
+    });
+    expect(handleDataModelChange).toHaveBeenCalled();
+  });
+
+  it('should render save icon', async () => {
+    await render();
+    const linkIcon = screen.getByText(/ux_editor.modal_properties_data_model_link/i);
+    act(() => {
+      linkIcon.click();
+    });
+    const saveButton = await screen.findByRole('button', { name: /general.save/i });
+    expect(saveButton).toBeInTheDocument();
+  });
+
+  it('should render delete icon', async () => {
+    await render();
+    const linkIcon = screen.getByText(/ux_editor.modal_properties_data_model_link/i);
+    act(() => {
+      linkIcon.click();
+    });
+    const deleteButton = await screen.findByRole('button', { name: /general.delete/i });
+    expect(deleteButton).toBeInTheDocument();
+  });
+
+  it('show link data model again when click on save button and no data model binding is selected', async () => {
+    await render();
+    const linkIcon = screen.getByText(/ux_editor.modal_properties_data_model_link/i);
+    act(() => {
+      linkIcon.click();
+    });
+    expect(
+      await screen.findByText(textMock('ux_editor.modal_properties_data_model_helper')),
+    ).toBeInTheDocument();
+    expect(screen.getByRole('combobox').getAttribute('value')).toEqual('');
+
+    const saveButton = await screen.findByRole('button', { name: /general.save/i });
+    act(() => {
+      saveButton.click();
```

Comment:
I see multiple places where the .click() on the element itself is used. Can you re-write them all to userEvent? 😊

Reply:
Fixed for all